lib/mri/monitor.rb

```diff
@@ -239,11 +239,9 @@ def initialize(*args)
   # Initializes the MonitorMixin after being included in a class or when an
   # object has been extended with the MonitorMixin
   def mon_initialize
-    unless @mon_mutex
-      @mon_owner = nil
-      @mon_count = 0
-      @mon_mutex = Mutex.new
-    end
+    @mon_owner = nil
+    @mon_count = 0
+    @mon_mutex = Mutex.new
```

Comment:
That was an intended change, missing a Truffle comment.
In the case mon_initialize is called twice but not concurrently, we do not want to overwrite the Mutex.
This happens in Rails tests.
I tried raising an exception instead but that unfortunately breaks many Rails tests.

Reply:
Will restore.